pkg/util/webhooks/validating-webhooks/validating-webhook.go

```diff
@@ -71,12 +72,28 @@ func Serve(resp http.ResponseWriter, req *http.Request, admitter admitter) {
 			Kind:       "AdmissionReview",
 		},
 	}
-	ctx := context.Background() // todo: use timeout context with the "timeout" request query parameter
+
+	ctx, cancel, err := getContextFromRequest(req)
+	if err != nil {
+		resp.WriteHeader(http.StatusBadRequest)
+		resp.Write([]byte(err.Error()))
+		return
+	}
+
+	defer func() {
+		// getContextFromRequest may return no cancel function, if the original context from the request already
+		// contains timeout.
+		if cancel != nil {
+			cancel()
+		}
+	}()
+
```

Comment:
Please consider the following alternative:
```suggestion
	timeout := 10 * time.Second
	if timeoutStr := req.URL.Query().Get("timeout"); timeoutStr != "" {
		if t, err := strconv.Atoi(timeoutStr); err == nil {
			timeout = time.Duration(t) * time.Second
		}
	}

	ctx, cancel := context.WithTimeout(req.Context(), timeout)
	defer cancel()
```

Reply:
why?

Reply:
It has less code branches, less error handling and does not require unit tests.

Reply:
I don't think so. The suggestion ignores the error, and this is wrong - so here is the missing branches. Also, the suggestion gives priority to the query parameter over the request context's timeout, and this is risky if it shorter than the query parameter. And finally, testable code is a good thing. unit test is a good thing. I'm confused by the last reason.

Reply:
Since the context with timeout is derived from the request's context, then if the parent context will be done, the derived context with timeout will be done as well.

This means that if either the request's context is done or the timeout has expired - the behavior is similar.